cmd/web/handlers.go

```diff
@@ -53,5 +48,84 @@ func (app *application) getCodeSystemByOID(w http.ResponseWriter, r *http.Reques
 
 	w.Header().Set("Content-Type", "application/json")
 
-	json.NewEncoder(w).Encode(codeSystems)
+	json.NewEncoder(w).Encode(codeSystem)
+}
+
+func (app *application) getAllViews(w http.ResponseWriter, r *http.Request) {
+	views, err := models.GetAllViews(r.Context(), app.db)
+	if err != nil {
+		if errors.Is(err, models.ErrDoesNotExist) {
+			http.NotFound(w, r)
+		} else {
+			app.serverError(w, r, err)
+		}
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+
+	json.NewEncoder(w).Encode(views)
+}
+
+func (app *application) getViewByID(w http.ResponseWriter, r *http.Request) {
+	id := r.PathValue("id")
+
+	view, err := models.ViewByID(r.Context(), app.db, id)
+	if err != nil {
+		if errors.Is(err, models.ErrDoesNotExist) {
+			http.NotFound(w, r)
+		} else if errors.Is(err, sql.ErrNoRows) {
+			errorString := fmt.Sprintf("Error: Code System %s not found", id)
```

Comment:
maybe update the error here to mention views instead of code systems

Reply:
Good catch!